providers/session_state.go

```diff
@@ -172,6 +185,20 @@ func DecodeSessionState(v string, c *cookie.Cipher) (*SessionState, error) {
 			User:  ss.User,
 		}
 	} else {
+		// Backward compatibility with using unecrypted Email
+		if ss.Email != "" {
+			decryptedEmail, errEmail := c.Decrypt(ss.Email)
```

Comment:
Is there any chance this `errEmail` could be `nil` whilst the original email wasn't encrypted? Just wondering if somehow `foo@example.com` could get garbled into `fnduoiafjd` by the decryption algorithm?

Reply:
No, is not. 
This scenario is covered by a unit test, the last test case from `TestDecodeSessionState`.
```
{
	SessionState: SessionState{
		Email:        "user@domain.com",
		User:         "just-user",
		AccessToken:  "token1234",
		IDToken:      "rawtoken1234",
		ExpiresOn:    e,
		RefreshToken: "refresh4321",
	},
	Encoded: fmt.Sprintf("email:user@domain.com user:just-user|I6s+ml+/MldBMgHIiC35BTKTh57skGX24w==|xojNdyyjB1HgYWh6XMtXY/Ph5eCVxa1cNsklJw==|%d|qEX0x6RmASxo4dhlBG6YuRs9Syn/e9sHu/+K", eUnix),
	Cipher:  c,
},
```
Trying to decrypt the email will make `errEmail` be `errorString{s:"failed to decrypt cookie value illegal base64 data at input byte 4"}`.

Reply:
Excellent, thanks for verifying 😄